### ISSUE TITLE
Figure 6 in the first paper

### ISSUE THREAD

**Issue by nguyenhongson1902** [thiswinex/STBP-simple#3]
Has anyone know what tools are used to generate the average number of spikes per neuron in each class? Thank you.
(As shown in figure 6 in the paper Going Deeper With Directly-Trained Larger Spiking Neural Networks)
By the way, can you show the way to compute the number of additions and multiplications like in Table 6?

**Comment by thiswinex** [comment]
I'm not the author of the paper so I can't provide their solutions. I think you can count the spike by customizing pytorch layer (LIF spike layer specially).

**Comment by nguyenhongson1902** [comment]
> I'm not the author of the paper so I can't provide their solutions. I think you can count the spike by customizing pytorch layer (LIF spike layer specially).

By the way you are here. Can you point me where to find the raw data for N-MNIST dataset, please?

**Comment by thiswinex** [comment]
> > I'm not the author of the paper so I can't provide their solutions. I think you can count the spike by customizing pytorch layer (LIF spike layer specially).
> 
> By the way you are here. Can you point me where to find the raw data for N-MNIST dataset, please?

I guess that's what you need: https://www.garrickorchard.com/datasets/n-mnist

**Comment by nguyenhongson1902** [comment]
I see that the function `adjust_learning_rate` is not used. Can you show me a simple way of implementing this function in your code base?

**Comment by thiswinex** [comment]
> I see that the function `adjust_learning_rate` is not used. Can you show me a simple way of implementing this function in your code base?

You can call this function in `train()` every epoch. However  `torch.optim.lr_scheduler` is recommended.

**Comment by nguyenhongson1902** [comment]
Have you tried running resnet19 built in your code with MNIST, N-MNIST, Fashion MNIST and CIFAR 10? With GPUs,  I've tried testing on MNIST  with resnet19 and it resulted in around 97 - 98 % accuracy, higher than the MNISTNet example (95%, without tdBN). But with CIFAR 10 dataset, I've tested with step 2 and resnet19 and the result is not good. It's about 70 % accuracy after 60 epochs and I see that the accuracy tends not to increase any more. That's only for step 2, but when increasing more steps (like 4, 6), I've found that it's a long time of training (a few days long) and the accuracy can't change much after nearly 100 epochs (the accuracy is around 44%). With those problems when training, I don't know how the authors got 93.15 % on CIFAR 10 with only 6 timesteps. About N-MNIST, I've also run with resnet19 and step 2. After 10 epochs, it seems like being stuck at 22-24 % accuracy. Can you have some advice for me to fix these problems? What did I do wrong when doing that procedures or if you have the results for those dataset, can you show me that? I really appreciate what you've told me.

**Comment by thiswinex** [comment]
I've only tried resnet19 on cifar10. It converged very slowly. I'm not sure if it will finally reach the line that the authors claimed  but you should keep training until it reaches 500 epoch or more if you want to get a better result (at least 75% acc).
